docs/core/runtime-discovery/troubleshoot-app-launch.md

```diff
@@ -0,0 +1,157 @@
+
+The following frameworks were found:
+  6.0.2 at [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
+```
+
+::: zone-end
+
+The error indicates the name, version, and architecture of the missing framework and the location at which it is expected to be installed. In order to run the application, you can [install a compatible framework](#install-a-compatible-framework) at the specified ".NET location". If the application is targeting a lower version than one you have installed and you would like to run it on a higher version, you can also [configure roll-forward behavior](#configure-roll-forward-behavior) for the application.
+
+### Install a compatible framework
+
+The error message includes a link to download the missing framework. You can follow this link to get the appropriate download page. Alternately, the [.NET downloads](https://dotnet.microsoft.com/download/dotnet) page offers all available versions of .NET. Select the version matching the one listed in the error message.
+
+Every version of .NET offers three different runtime downloads. The table below shows the frameworks contained by each of them.
+
+| Download name        | Included frameworks |
+| -------------------- | ------------------- |
+| ASP.NET Core Runtime | Microsoft.AspNetCore.App<br/>Microsoft.NETCore.App |
+| .NET Desktop Runtime | Microsoft.NETCore.App<br/>Microsoft.WindowsDesktop.App |
+| .NET Runtime         | Microsoft.NETCore.App |
+
+To install a compatible framework, on the download page for the required .NET version, find a runtime download containing the missing framework. Once you have found the appropriate runtime download, you can then either install it using an [installer](#run-an-installer) or [download the binaries](#download-binaries) and extract them to the desired location.
+
+#### Run an installer
+
+If your existing installation of .NET was installed through an installer or package manager, then also installing the required framework through an installer or package manager is likely the simpler option. Otherwise, you can [download binaries](#download-binaries) instead of using an installer.
+
+In most cases, when the application that failed to launch is using such an installation, the ".NET location" in the error message would point to:
+::: zone pivot="os-windows"
+`%ProgramFiles%\dotnet`
+::: zone-end
+::: zone pivot="os-linux"
+`/usr/share/dotnet/`
+::: zone-end
+::: zone pivot="os-macos"
+`/usr/local/share/dotnet/`
+::: zone-end
+
+::: zone pivot="os-windows,os-macos"
+From the **Installers** column of the runtime download, download the installer matching the required architecture. Run the downloaded installer.
+::: zone-end
+::: zone pivot="os-linux"
+Different Linux distributions provide .NET through different package managers. See [Install .NET on Linux](../install/linux.md) for details. Note that preview versions of .NET are not available through package managers.
+::: zone-end
+
+#### Download binaries
```

Comment:
@richlander - not sure if we want to promote extracting zips as an install experience. Maybe this should point to some overall "how to install" page just noting the typical difference between default location (should use installers/packages) and non-default location in which case I would recommend usage of the `dotnet-install` script.

Reply:
I kind of agree - but to a point. I think the general problem with the current web-site/download/install experience as driven by the host is that we don't point to a specific single download - the user still has to choose (click on the right option on the web-site).

The experience for the installers is reasonable in that the web-site has nice, large pictures and text and so on, and it takes like 1-2 clicks to get the thing (assuming you know what - which is another problem).

Downloading the zips is worse - you have to go through the a "All downloads" page which is overwhelming even for me. In that sense, asking the user to run something like `dotnet-install.ps1 --version 1.2.3 --framework ASP.NET --location C:\private` feels much simpler and also feels much more in-line with the CLI experience which is probably the preferred way for devs anyway. And it's fully scriptable/repeatable (as in - oh I need new SDK for this thing, OK, now it works locally, but what do I add to my CI to make it work there).

But I agree that this is a much larger discussions and should not block these changes in any way.

Reply:
That's definitely fair. 'All downloads' does have a lot going on and the user needs to map the information in the error message to multiple separate actions/clicks, while the script can just take all the information in a single command.

I'm going to put this in its current state so that we at least have a page. I'll likely add a section pointing at the script later. (And I'm sure we will need some iteration / improvement after.)